lib/spack/spack/modules/common.py

```diff
@@ -774,23 +774,29 @@ def conflicts(self):
         """List of conflicts for the module file."""
         fmts = []
         projection = proj.get_projection(self.conf.projections, self.spec)
-        f = string.Formatter()
         for item in self.conf.conflicts:
-            if len([x for x in f.parse(item)]) > 1:
-                for naming_dir, conflict_dir in zip(projection.split("/"), item.split("/")):
-                    if naming_dir != conflict_dir:
-                        message = "conflict scheme does not match naming "
-                        message += "scheme [{spec}]\n\n"
-                        message += 'naming scheme   : "{nformat}"\n'
-                        message += 'conflict scheme : "{cformat}"\n\n'
-                        message += "** You may want to check your "
-                        message += "`modules.yaml` configuration file **\n"
-                        tty.error(message.format(spec=self.spec, nformat=projection, cformat=item))
-                        raise SystemExit("Module generation aborted.")
-                item = self.spec.format(item)
+            self._verify_conflict_naming_consistency_or_raise(item, projection)
+            item = self.spec.format(item)
             fmts.append(item)
-        # Substitute spec tokens if present
-        return [self.spec.format(x) for x in fmts]
+        return fmts
+
+    def _verify_conflict_naming_consistency_or_raise(self, item, projection):
+        f = string.Formatter()
+        errors = []
+        if len([x for x in f.parse(item)]) > 1:
+            for naming_dir, conflict_dir in zip(projection.split("/"), item.split("/")):
+                if naming_dir != conflict_dir:
+                    errors.extend(
+                        [
+                            f"spec={self.spec.cshort_spec}" f"conflict_scheme={item}",
```

Comment:
Isn't there a missing `,` between f"spec*" and f"conflict_scheme*"?

Error is hard to read otherwise:

```
E           spack.modules.common.ModulesError: conflict scheme does not match naming scheme
E               spec=mpileaks@2.3%gcc@10.2.1~debug~opt+shared+static build_system=generic arch=test-debian6-core2/be7geaaconflict_scheme={name}/{compiler.name}
E           naming_scheme={name}/{version}-{compiler.name}
```

Reply:
There is indeed!

Reply:
Fixed